clients/da-vinci-client/src/main/java/com/linkedin/davinci/schema/merge/SortBasedCollectionFieldOpHandler.java

```diff
@@ -51,6 +51,12 @@ public UpdateResultStatus handlePutList(
       toPutList = Collections.emptyList();
     } else {
       toPutList = newFieldValue;
+      /**
+       * LinkedList is more efficient for the following add/remove operations.
+       */
+      if (!(toPutList instanceof LinkedList)) {
+        toPutList = new LinkedList<>((toPutList));
+      }
```

Comment:
Is it possible to deserialize directly into a `LL` rather than copying here?

Reply:
List optimization is quite complex.
I will update the code to only copy `LL` when the collection is not in a `put_only` state to reduce unnecessary `LL` copy operations.

Today, for `list` type, there are following requirements:
1. For the list type in `value`, it requires `LL` related ops, such as `remove`.
2. For the list type in `rmd`, it requires fast lookup.
3. In fast-avro, for the primitive list, we are using an optimized structure to store them to reduce deserialization/serialization effort.

Optimizing every scenario is quite hard, and I would like to merge this PR and run another round of profiling before further List optimization because of the complexity.
What do you think?

Reply:
IIUC, 1 and 2 are different payloads, which can be deserialized with different deserializers, each of which we can configure to use a different type of list thanks to the recent fast-avro improvement, right?

Regarding 3, it is the `rmd` that will have a list of `long`, but the value could be a list of anything, not just primitives, right? So it should be fine to use fast-avro's default primitive lists for use case 2.

I don't mind us doing it incrementally across several PRs. I'm just trying to understand what is feasible.

Reply:
yes, but for 1, it can contains primitive arrays, such as embeddings, and if we blindly deserialize them into LinkedList, we will lose the benefit of #3, which will save cpu and GC effort.

Let us profile it again after deploying this change to prod.

Reply:
I see.